daemon.go

```diff
@@ -48,6 +49,12 @@ type Daemon struct {
 	// callID (int64) to chan context.CancelFunc
 	// used to cancel request handlers
 	cancelUnary sync.Map
+
+	// this sync.Once ensures the goroutine awaiting deamon termination is
+	// only run once
+	terminateOnce   sync.Once
+	terminateWG     sync.WaitGroup
+	cancelTerminate context.CancelFunc
```

Comment:
```suggestion
	cancelTerminateTimer context.CancelFunc
```

Let's make clear that it does not cancel all ways to terminate the daemon (like SIGTERM), but cancels only the timer :)